### ISSUE TITLE
Doesn't call #close on the Rack response body when it's empty

### ISSUE THREAD

**Issue by janko-m** [socketry/http-protocol#3]
Steps to reproduce:

```rb
# config.ru
run -> (env) {
  body = Rack::BodyProxy.new([]) { puts "This is never called" }

  [200, {}, body]
}
```
```
$ falcon serve -b http://localhost:9292
```
```
$ curl localhost:9292
```

I expected to see `This is never called` in the server output, but it's not there, which means Falcon never called `#close` on the response body. Puma call `#close` in this case, and if response body is nonempty then Falcon does as well.

This behaviour breaks some middlewares, for example `Rack::CommonLogger` won't log requests that have empty response bodies, because the logging is done on `#close` (it also uses `Rack::BodyProxy`).

**Comment by ioquatix** [comment]
I'm working on a better fix. The requirement to call each is a bit silly.

**Comment by janko-m** [comment]
@ioquatix Thanks a lot for the fix!

**Comment by ioquatix** [comment]
Yeah, I released a quick fix, but it seems logical to handle this a bit higher up rather than depending on the protocol to do this.

**Comment by ioquatix** [comment]
Okay, I released falcon v0.19.4 which has a better implementation handling body `#close`.
